py-polars/tests/unit/io/test_csv.py

```diff
@@ -1710,3 +1710,10 @@ def test_csv_no_new_line_last() -> None:
         "a": [1, 2, 3],
         "b": [1.0, 2.0, 2.1],
     }
+
+
+def test_write_csv_no_location_raise_io_exception() -> None:
+    df = pl.DataFrame({"a": [1]})
+    # TODO: check dir don't exists before writing
+    with pytest.raises(IOError):
```

Comment:
We should probably extend this list to get it working properly:

https://github.com/pola-rs/polars/blob/026377c45c04cfe225f508d53750cef79ef88af6/py-polars/src/error.rs#L42-L47

Reply:
1. Why? According by python official docs the [FileExistsError](https://docs.python.org/3/library/exceptions.html#FileExistsError) is for when "trying to create a file or directory which already exists."

2. I would love some guidance on how to control the exact Python error from the rust code. I was only able to wrap it in a `PolarsError::IO` that ended up as a `FileNotFoundError`.

Reply:
1. I was mistaken. It should be `FileNotFoundError`, because it is a case where the "directory is requested but doesn’t exist".
2. Even if we want another error, the io Rust standard lib throws a `NotFound` for both cases. See, please: https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#errors

> [NotFound](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.NotFound): The specified file does not exist and neither create or create_new is set.
[NotFound](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.NotFound): One of the directory components of the file path does not exist.

If you still think we need another error type to be raised and you know how to differentiate in rust by the two same error classes, I would like to know.

Reply:
Ok. Thanks. 

May I use this conversation and kindly ask you to explain why you guys wrap all errors in `PolarsError` and `PyPolarsError` just to unwind them back to Python errors?

Reply:
Only some errors get converted to Python errors. I believe `IOError` is the exception here.

If you have a better way of doing things, I'd be happy to hear it.

Reply:
Thanks. 😊